### PR TITLE
HomewordFour

### PR DESCRIPTION
**Pull request by YauheniMakarenko**

### CODE REVIEW

File: src/HomeworkFour/Figure.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package HomeworkFour;
+
+public abstract class Figure {
+
+    public abstract String getName();
+    public abstract double periment();
+    public abstract double area();
+    public abstract String toString();
```
Comment (vitalyahacker) [inline src/HomeworkFour/Figure.java]: toString is already inherited from Object, you don't need to add it

File: src/HomeworkFour/Figure.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package HomeworkFour;
+
+public abstract class Figure {
+
+    public abstract String getName();
```
Comment (vitalyahacker) [inline src/HomeworkFour/Figure.java]: name is a field in figure class, getter should be only in figure class, not abstract

File: src/HomeworkFour/Rectangle.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    public void setWidth(int width) {
+        this.width = width;
+    }
+
+    public int getWidth(){
+        return width;
+    }
+
+    @Override
+    public String getName() {
```
Comment (vitalyahacker) [inline src/HomeworkFour/Rectangle.java]: don't need to override this method

File: src/HomeworkFour/Ships/Ship.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
+package HomeworkFour.Ships;
+
+import java.util.Random;
+
+public class Ship {
+    int size;
```
Comment (vitalyahacker) [inline src/HomeworkFour/Ships/Ship.java]: why all fields have default access modifier?

File: src/HomeworkFour/Ships/Ship.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    int positionX;
+    int positionY;
+    int rnd;
+    Random random = new Random();
```
Comment (vitalyahacker) [inline src/HomeworkFour/Ships/Ship.java]: should be constant, but not field

File: src/HomeworkFour/Sphere.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+    public void setRadius(int radius){
+        this.radius = radius;
+    }
+    public int getRadius(){
+        return radius;
+    }
+
+    @Override
+    public String getName() {
```
Comment (vitalyahacker) [inline src/HomeworkFour/Sphere.java]: do not override the method

File: src/HomeworkFour/Square.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    public void setLength(int length) {
+        this.length = length;
+    }
+
+    public int getLength(){
+        return length;
+    }
+
+    @Override
+    public String getName(){
```
Comment (vitalyahacker) [inline src/HomeworkFour/Square.java]: do not

File: src/HomeworkFour/Ships/Cell.java [hunk 8]
```diff
@@ -0,0 +1,93 @@
+package HomeworkFour.Ships;
+
+public class Cell {
```
Comment (vitalyahacker) [inline src/HomeworkFour/Ships/Cell.java]: Cell is ячейка 
I believe that it's not the best name to it

File: src/HomeworkFour/Picture.java [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+    public String namesOfFigures() {
+        String sum = "";
+        for (int i = 0; i < list.size(); i++) {
+            sum += list.get(i).getName() + " ";
+        }
+        return sum;
+    }
+
+    public Map countDifferentFigures() {
```
Comment (v-bliakharchu-idf) [inline src/HomeworkFour/Picture.java]: do not use map without type parameters

File: src/HomeworkFour/Ships/Field.java [hunk 10]
```diff
@@ -0,0 +1,90 @@
+package HomeworkFour.Ships;
+
+public class Field {
+    static int SIZE = 10;
+    private char[][] cell = new char[SIZE][SIZE];
+    private final int[] DECK = {4, 3, 3, 2, 2, 2, 1, 1, 1, 1};
+    Ship[] ships = new Ship[DECK.length];
```
Comment (v-bliakharchu-idf) [inline src/HomeworkFour/Ships/Field.java]: why fields without access modifiers?

File: src/HomeworkFour/Ships/Field.java [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+        if (tmpShip.getRnd() == 0) {
+            for (int i = 0; i < tmpShip.getSize(); i++) {
+                cell[tmpShip.getPositionY() + i][tmpShip.getPositionX()] = '#';
+            }
+        } else {
+            for (int i = 0; i < tmpShip.getSize(); i++) {
+                cell[tmpShip.getPositionY()][tmpShip.getPositionX() + i] = '#';
+            }
+        }
+    }
```
Comment (v-bliakharchu-idf) [inline src/HomeworkFour/Ships/Field.java]: why methods withous access modifiers?

File: src/HomeworkFour/Ships/Ship.java [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+
+    public int getRnd() {
+        return rnd;
+    }
+
+    public void setRnd(int rnd) {
+        this.rnd = rnd;
+    }
+
+    void initWithRandom() {
```
Comment (v-bliakharchu-idf) [inline src/HomeworkFour/Ships/Ship.java]: access modifier

File: src/HomeworkFour/Ships/Ship.java [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+import java.util.Random;
+
+public class Ship {
+
+    private int size;
+    private int positionX;
+    private int positionY;
+    private int rnd;
+
+    Ship(int Deck) {
```
Comment (v-bliakharchu-idf) [inline src/HomeworkFour/Ships/Ship.java]: access modifier

File: src/HomeworkFour/Ships/Main.java [hunk 14]
```diff
@@ -0,0 +1,11 @@
+package HomeworkFour.Ships;
+
+public class Main {
+    public static void main(String[] args) {
+        Field field = new Field();
+        field.fillField();
+        field.shipOnTheField();
+        field.viewField();
```
Comment (v-bliakharchu-idf) [inline src/HomeworkFour/Ships/Main.java]: bad names, it's not obvious what they do

**Pull request merged**